scripts/zenodo-bibtex.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/python
+"""This script forms part of the internal workings of Firedrake-Zenodo. It is not intended for direct user interaction."""
+import cgi
+import cgitb
+cgitb.enable()
+import urllib2
+import json
+from HTMLParser import HTMLParser
+import sys
+
+
+def fail(code, reason):
+
+    sys.stdout.write('Status: %s\r\n\r\n%s\r\n\r\n' % (code, reason))
+    sys.exit(1)
+
+def find_ids(response, firedrake_tag):
+
+    ids = []
+    for record in response:
+        try:
+            for id in record['metadata']['related_identifiers']:
+                if id['identifier'].endswith(firedrake_tag):
+                    ids.append(record['record_id'])
+        except KeyError:
+            pass
+
+    return ids
+
+
+class BibtexFinder(HTMLParser):
+    """The BibTeX data is kept inside tag <pre id="clipboard_text">."""
+    def __init__(self):
+        HTMLParser.__init__(self)
+
+        self.in_bibtex = False
+        self.bibtexdata = None
+
+    def handle_starttag(self, tag, attrs):
+        if tag == "pre":
+            a = dict(attrs)
+            if a.get("id") == "clipboard_text":
+                self.in_bibtex = True
+
+    def handle_data(self, data):
+        if self.in_bibtex:
+            self.bibtexdata = data
+
+    def handle_endtag(self, tag):
+        self.in_bibtex = False
+
+
+def get_bibtex(ids):
+
+    bibtex = []
+    for id in ids:
+        html = urllib2.urlopen("https://zenodo.org/record/%s/export/hx" % id)
+        parser = BibtexFinder()
+        parser.feed(html.read())
+        if parser.bibtexdata:
+            bibtex.append(parser.bibtexdata)
+
+    return "\n\n".join(bibtex)
+
+
+form = cgi.FieldStorage()
+firedrake_tag = form.getfirst("tag")
+if not firedrake_tag:
+    fail("400 Bad Request", "Request must specify the tag.")
+# Do some primitive parsing to avoid more blatant security issues.
+try:
+    assert firedrake_tag.startswith("Firedrake_")
+    float(firedrake_tag[10:])
```

Comment:
I need to be able to cast the remainder of the tag to a `float`.  We pass the entire rest of the user-provided string to `float`, so a malformed input *could* expose a bug in core python.  We know something more about the tag, right.  If it's a valid tag, it must be `YYYYMMDD.NNNN`. So we could do:

```python
date = int(firedrake_tag[10:18])
assert tag[18] == "."
# assume we will not create more than 9999 tags on the same commit
number = tag[19:23]
```